src/services/textChanges.ts

```diff
@@ -365,6 +388,36 @@ namespace ts.textChanges {
             this.insertNodeAt(sourceFile, endNode.end, type, { prefix: ": " });
         }
 
+        public tryInsertJSDocType(sourceFile: SourceFile, node: Node, type: TypeNode): void {
+            const printed = createPrinter().printNode(EmitHint.Unspecified, type, sourceFile);
+            let commentText;
+            if (isGetAccessorDeclaration(node)) {
+                commentText = ` @return {${printed}} `;
+            }
+            else {
+                commentText = ` @type {${printed}} `;
+                node = node.parent;
+            }
+            this.insertCommentThenNewline(sourceFile, getLineAndCharacterOfPosition(sourceFile, node.getStart()).character, node.getStart(), commentText);
+        }
+
+        /** Should only be used to repeat strings a small number of times */
+        private repeatString(s: string, n: number) {

```

Comment:
This doesn't really belong as an instance method. There's also already a function with this name.

Reply:
Oops, I forgot about services/utilities.ts and only looked in compiler/utilities.ts